Add Avali From RMC-14

Forzii opened the pull request:
<!--
This is a semi-strict format, you can add/remove sections as needed but the order/format should be kept the same
Remove these comments before submitting
-->

# Description

<!--
Explain this PR in as much detail as applicable

Some example prompts to consider:
How might this affect the game? The codebase?
What might be some alternatives to this?
How/Who does this benefit/hurt [the game/codebase]?
-->

Description.

---

# TODO

<!--
A list of everything you have to do before this PR is "complete"
You probably won't have to complete everything before merging but it's good to leave future references
-->

- [x] https://github.com/RMC-14/RMC-14/pull/4336
- [x] Add Avali to undershirt and underwear markings
- [x] Move Markings avali.yml to the right folder (cherry-pick being weird)
- [x] Reparent the avali entity
- [x] de-duplicate conflicting entities (voice sounds, etc)
- [x] fix markings
- [ ] (optional) de-duplicate anything not conflicting we already have for Resomi
- [ ] (optional) parent or somehow merge resomi and avali? I tried making Resomi a parent for Avali and it mostly worked - running into some issues with displacement maps
- [ ] (optional) add species stuff? Like... min/max height, density, weapon damage tweak, ammonia blood, etc?
- [x] https://github.com/RMC-14/RMC-14/pull/4355 (didn't need, not relevant to us)
- [x] https://github.com/RMC-14/RMC-14/pull/4455
- [x] https://github.com/RMC-14/RMC-14/pull/4677
- [x] https://github.com/RMC-14/RMC-14/pull/4768
- [ ] https://github.com/RMC-14/RMC-14/pull/4845 (Seems this might be a tricky one)
- [x] https://github.com/RMC-14/RMC-14/pull/5503
- [x] https://github.com/RMC-14/RMC-14/pull/5974 (includes other changes that need to be discarded)
- [x] Remove/replace noSpawn: true in MobAvaliDummy
- [x] Figure out what save: false in MobAvali is for and if it's relevant/problematic here (I dunno what it does but Resomi has it too so it's prolly fine)
- [x] Fix Avali random name generation (I see the issue. We didn't do the names switch to localizedDataset like wizden did. I'll have to undo the switch RMC did but keep the extra names they added.)
- [ ] Compare with Resomi/others to see what components are missing (LanguageKnowledge, FootPrints, what else?)
- [ ] Test things (sounds, typing animation, general emotes, markings, damage, etc)
- [ ] Finish PR proper

---

<!--
This is default collapsed, readers click to expand it and see all your media
The PR media section can get very large at times, so this is a good way to keep it clean
The title is written using HTML tags
The title must be within the <summary> tags or you won't see it
-->

<details><summary><h1>Media</h1></summary>
<p>

![Example Media Embed](https://example.com/thisimageisntreal.png)

</p>
</details>

---

# Changelog

<!--
You can add an author after the `:cl:` to change the name that appears in the changelog (ex: `:cl: Death`)
Leaving it blank will default to your GitHub display name
This includes all available types for the changelog
-->

:cl:
- add: Added fun :D
- tweak: Tweaked fun
- fix: Fixed fun!
- remove: Removed fun :(

Vermidia committed:
Add Avali (#4336)

Vermidia committed:
Add avali speech sounds (#4455)

crystalHex committed:
Fix typo in Avali Urist prototype name (#4677)

blueDev2 committed:
Adjust Avali Sprites (#4768)

Co-authored-by: DrSmugleaf <10968691+DrSmugleaf@users.noreply.github.com>

Vermidia committed:
Add Avali laugh (#5503)

DrSmugleaf committed:
Fix species names and add 58 more Avali names (#5974)

Forzii committed:
Some merge cleanup

Forzii committed:
cherry-pick cleanup

Removed text that wasn't added by the PR

Forzii committed:
Deduplicate sounds, fix marking categories

Forzii committed:
Merge remote-tracking branch 'upstream/master' into add-avali

Forzii committed:
Replace noSpawn

Forzii committed:
Fix avali prototype name

ShadesMars commented:
Resomi are not the same as such?

Forzii committed:
Revert localizedDataset

We haven't switched!

M3739 commented on Resources/Prototypes/_RMC14/Voice/Avali/avali_emote_sounds.yml:
```diff
@@ -0,0 +1,15 @@
+
+- type: emoteSounds
+  parent: [RMCUnisexAvali, RMCFemaleHuman]
+  id: RMCFemaleAvali
```
Our current implementation of the `EmoteSoundsPrototype.cs` does not feature parenting.

In addition, we already have a file with the same name located in our folder. It would be desirable to take the necessary additions and changes to our file instead.

For reference, here is the github link that points to our [`avali_emote_sounds.yml`](https://github.com/Fansana/floofstation1/blob/master/Resources/Prototypes/_Floof/Voice/avali_emote_sounds.yml).

Forzii replied:
Ah, yeah. I think I already de-conflicted the sounds ( need to double-check) but apparently forgot to update the paths. Thanks!

M3739 commented on Resources/Prototypes/_RMC14/Entities/Mobs/Species/avali.yml:
```diff
@@ -0,0 +1,36 @@
+    sounds:
+      Male: RMCMaleAvali
+      Unsexed: RMCMaleAvali
+      Female: RMCFemaleAvali
```
You may want to update these once you merge the changes and additions to our avali emote sounds.

M3739 commented on Resources/Prototypes/_RMC14/Datasets/avali_names.yml:
```diff
@@ -0,0 +1,81 @@
+    - Naoli
+    - Kazala
+    - Ruli
+    - Jesi
+    - Einuni
+    - Eija
+    - Kazumi
+    - Nuni
+    - Kalomi
+    - Jasun
+    - Ranoli
+    - Nahumi
+    - Eikoli
+    - Einyo
+    - Ralo
+    - Nezuko
+    - Kalako
+    - Naaka
+    - Naani
+    - Nahuli
+    - Eiluno
+    - Nezoli
+    - Einomi
+    - Kaka
+    - Nesi
+    - Eijali
+    - Nahi
+    - Einun
+    - Nao
+    - Nezala
+    - Nuno
+    - Halako
+    - Nahali
+    - Neza
+    - Nahan
+    - Talo
+    - Eitaka
+    - Jesumi
+    - Ranun
+    - Jesali
```
The Resomi have their own namelist to themselves, both of which are in our folder, and both of which are split into their gendered counterparts: [`resomi_male.yml`](https://github.com/Fansana/floofstation1/blob/master/Resources/Prototypes/_Floof/Datasets/Names/resomi_male.yml) and [`resomi_female.yml`](https://github.com/Fansana/floofstation1/blob/master/Resources/Prototypes/_Floof/Datasets/Names/resomi_female.yml).

It may be in your best interest to take these names, as they are unisex according to the title, and combine them into those two lists, and instead make the species prototype bear the same `maleFirstNames` and `femaleFirstNames` fields.

Forzii replied:
I'm no expert on naming. I'm thinking we have a few options... make an unsexed dataset for unsexed characters using the RMC ones, combine them into both male and female (like you suggested), or perhaps enlist help to 'gender' the names. Obviously the first two would be easier... I did a diff and it seems like the names are entirely unique from each other.

M3739 reviewed (changes requested):
Only major issue is the `emoteSounds` being erroneous for our codebase. Other than that, largely suggestions regarding possibly merging the name dataset and emote sounds into what we already got for the Resomi.

M3739 commented:
Just checking in. How's the progress so far?

Forzii commented:
> Just checking in. How's the progress so far?

Blehhh, sorry. I'll working on it some today. My motivation has been shot lately. Also seems like every time I look at SS14 stuff there's constant drama and complaining, which doesn't help. I'll work on it some today.

Forzii committed:
Combine avali and resomi noises

I only did what was necessary to stop conflict errors before, this is a bit more comprehensive hopefully.

M3739 commented:
> > Just checking in. How's the progress so far?
> 
> Blehhh, sorry. I'll working on it some today. My motivation has been shot lately. Also seems like every time I look at SS14 stuff there's constant drama and complaining, which doesn't help. I'll work on it some today.

All good. Take your time, and no rush.

Forzii committed:
Merge remote-tracking branch 'upstream/master' into add-avali

Forzii committed:
Use Resomi as parent

This mostly works, but there's some weirdness with displacement maps stuff.
